### ISSUE TITLE
SSH e2e test is flaky on GKE

### ISSUE THREAD

**Issue by quinton-hoole** [GoogleCloudPlatform/kubernetes#7714]
On May 3, 2015 7:36 PM, "Brendan Burns" <bburns@google.com> wrote:
56% successful:

SSH_should_SSH_to_all_nodes_and_run_commands

My guess is SSH isn't coming up fast enough?  But that's just a guess.

Can someone take a look on Monday?

Thanks!
--brendan

**Comment by roberthbailey** [comment]
The failures take less time that the successful runs, which is usual for our e2e tests. 

The output from failing runs looks like: `error getting SSH client to host 104.154.56.160:22: 'ssh: handshake failed: ssh: unable to authenticate, attempted methods [none publickey], no supported methods remain'`.

**Comment by mbforbes** [comment]
I'll check this out. It would make sense if it was "SSH not coming up fast enough" (though: what does that mean? the ssh daemon on the nodes?) because I never saw a single failure while running a few dozen times locally on a cluster that had been up for a while.

**Comment by mbforbes** [comment]
I ran the following command to create a cluster and immediately run the SSH test on it:

```bash
go run hack/e2e.go --v --up --test --test_args="--ginkgo.focus=SSH"
```

and the test passes. This makes me wonder whether it might be related to the SSH key used in Jenkins. I've created #7732 to try to help debug this.

**Comment by roberthbailey** [comment]
Jenkins is certainly cycling SSH keys very rapidly.

**Comment by mbforbes** [comment]
Yes... in fact, I wonder now, even if it is using the right key, whether the key might have been replaced with the wrong content. The test would have errored out a few lines before if the key was gone.

**Comment by mbforbes** [comment]
Just for the record—and for those prone to skimming over a character—this test has never failed (not even once) on GCE; it is only problematic on GKE.

**Comment by mbforbes** [comment]
For anyone reading along, we think this is an issue with GCP and SSH keys. We think we're hitting the limit of the number of SSH keys we can have saved in GCP for the Jenkins project; each new build creates a new key and recycles other keys, which can cause other builds to fail (the SSH test) when their keys get removed.

We're confirming this now, and will then implement a client-side fix (just using a few "permanent" keys) on our Jenkins box if this is true.

**Comment by mbforbes** [comment]
Woohoo! This was fixed by #8004 (Jenkins was red for a while so it was hard to confirm).
